docs/technical-design/howto-health-plan-form-data-changes.md

```diff
@@ -0,0 +1,34 @@
+# How to handle changes in the Health Plan Form Data
+
+## Summary
+
+| Data Layer     | Adding a field | Removing a field | Changing the type of a field |
+| ----------- | ----------- | ----------- | ----------- |
+| Protocol Buffers    |  Straightforward, just add it!   | Deprecate, don't delete. | Go for it but watch out for data loss.
+| Domain Models   | Use caution for required fields, you will need to handle fallbacks for old data. Straightforward for optional fields.     | Deletion is okay but deprecate if product wants to display this data to users on old submissions. | Use caution here, you will need to handle fallbacks.
+
+## Further discussion
+
+### Protobufs
+
+Relevant filepath: 
+- [health_plan_form_data.proto](/services/app-proto/src/health_plan_form_data.proto)
+
+Many changes to the protobuf schema itself are straightforward because protos are flexible, have unique field numbers, and use optional field. This means that adding a field or deprecating a field (`[deprecated = true]`) does not effect the ability of the proto decoder. If a newer schema is used to read data written with an older schema, the proto decoder drops fields that are no longer in the schema. It also will not error for fields that are present in new schema but missing from old data (they ar just assigned to undefined or empty list).
```

Comment:
```suggestion
Many changes to the protobuf schema itself are straightforward because protos are flexible, have unique field numbers, and use optional fields. This means that adding a field or deprecating a field (`[deprecated = true]`) does not affect the ability of the proto decoder. If a newer schema is used to read data written with an older schema, the proto decoder drops fields that are no longer in the schema. It also will not error for fields that are present in the new schema but missing from old data (they are just assigned to undefined or empty list).
```